### PR TITLE
Issue #1936 - aliasQuery function

### PR DESCRIPTION
**Pull request by mageo**
Hello,

Following my feature request #1936, I had a look to the code and it appeared not so difficult to implement it by myself.

Here is the result.

I hope someone else may find this useful.

Code has been tested in production.

### CODE REVIEW

**Comment by deniszh** [comment]
Hello @mageo,
Thanks for your contribution, much appreciated!
But we have strong policy about testing functions - would you mind to add tests to your function?
You can check examples in webapp/tests/test_functions.py or check other PRs, e.g. https://github.com/graphite-project/graphite-web/pull/1928

**Comment by mageo** [comment]
Hello @deniszh,
I will have a look at the examples and write a test function.

**Comment by deniszh** [comment]
Looks good. Codacy is too picky for lambdas, can be ignored.
Let us review it.

File: webapp/graphite/render/functions.py [hunk 1]
```diff
@@ -1734,6 +1734,25 @@ def aliasSub(requestContext, seriesList, search, replace):
...
+
+  .. code-block:: none
+
+    &target=aliasQuery(channel.power.*,"channel\.power\.([0-9]+)","channel.frequency.\\1", "Channel \\1 MHz")
+  """
+  for serie in seriesList:
+    newQuery = re.sub(search, query, serie.name)
+    newSeriesList = evaluateTarget(requestContext, newQuery)
+    if newSeriesList is None or len(newSeriesList) == 0:
+      raise Exception('No serie found with query: ' + newQuery)
```
Comment (cbowman0) [inline webapp/graphite/render/functions.py]: Singular of series is still series.  At least for the error message, can you change it to be series?

Comment (cbowman0) [inline webapp/graphite/render/functions.py]: Also, I don't think this failure condition is tested in the tests provided.  Can you add another test or two to test the failure conditions?

The rest looks good!

Comment (mageo) [inline webapp/graphite/render/functions.py]: Sorry, series is similar in french and singular is serie. I will change that. I will also update the test function to test the failure conditions.

Comment (cbowman0) [inline webapp/graphite/render/functions.py]: Ah.  I thought it would be something like that.  English isn't always logical...

**Comment by deniszh** [comment]
Thanks, @mageo, merged!

**Comment by DanCech** [comment]
I was going to say that I like this a lot but feel that it'd be more powerful if it used sprintf syntax for building the alias vs regex replacement

**Comment by DanCech** [comment]
@deniszh as a more general comment, we need to be cognizant that once a new function is added to the API we're making a commitment to supporting it going forward and we should allow time for discussion to happen before we merge something like this.

**Comment by deniszh** [comment]
@DanCech :
Yes, you're right, I was too fast with merging that. Usually, I wait a couple of days before merging, but these was approved so fast and looks OK...
We can fix it, though - or even revert - it's still in master, which is subject to change.

Could you please elaborate what you mean "be more powerful if it used sprintf syntax for building the alias" btw?

**Comment by DanCech** [comment]
@deniszh cool, I'll throw together a PR to demo what I mean.

**Comment by mageo** [comment]
@DanCech yes, your approach would give way more flexibility !

Do you think something like this would be enough ?

`series.name = newName.format(current)`

**Comment by DanCech** [comment]
More or less, I have a patch that uses `series.name = newName % current` since that syntax is much closer to the standard c-style printf syntax.

https://docs.python.org/2/library/stdtypes.html#string-formatting-operations
https://en.wikipedia.org/wiki/Printf_format_string

I'm just expanding the docs a bit and I'll open a PR for that.

**Comment by DanCech** [comment]
#1938

**Pull request merged**